src/renderer/entities/price/model/__tests__/price-provider-model.test.ts

```diff
@@ -74,4 +74,13 @@ describe('entities/price/model/price-provider-model', () => {
     await allSettled(priceProviderModel.events.assetsPricesRequested, { scope, params: { includeRates: false } });
     expect(scope.getState(priceProviderModel.$assetsPrices)).toEqual(prices);
   });
+
+  test('should update asset prices on currency change', async () => {
+    jest.spyOn(fiatService, 'getAssetsPrices').mockReturnValue(prices);
+
+    const scope = fork();
+    await allSettled(kernelModel.events.appStarted, { scope });
+    await allSettled(currencyModel.events.currencyChanged, { scope, params: 1 });
+    expect(scope.getState(priceProviderModel.$assetsPrices)).toEqual(prices);
```

Comment:
I suppose we need to compare value before and value after `currencyModel.events.currencyChanged`
instead of checking `prices` after initial event (`kernelModel.events.appStarted`)
By the way, you can provide `$assetPrices` inside `fork` without emitting `appStarted` take a look at test located above `test('should fetch $assetsPrices when assetsPricesRequested')`